### PR TITLE
Limit the filesize of static code analysis reports

### PR DESCRIPTION
**Pull request by ivanchimeno**
The PR limits the amount of reports which can be parsed by the static code analysis parser. The current limit is set to 1MB and returns an issue if the limit is surpassed.

This PR also refactors the code: 

- converts global class objects to local ones so that the garbage collector can clean them after use
- removes  support of parsing reports through `InputStream` because the size cannot be determined ahead of time.

### CODE REVIEW

File: src/main/java/de/tum/in/ase/parser/strategy/ParserContext.java [hunk 1]
```diff
@@ -2,23 +2,22 @@
...
 public class ParserContext {
 
-    private ParserStrategy parserStrategy;
-    private ParserPolicy parserPolicy = new ParserPolicy(this);
-
-    public void setParserStrategy(ParserStrategy parserStrategy) {
-        this.parserStrategy = parserStrategy;
-    }
+    // Reports that are bigger then the threshold will not be parsed
+    // and an issue will be generated. The unit is in bytes.
```
Comment (FrankeLukas) [inline src/main/java/de/tum/in/ase/parser/strategy/ParserContext.java]:
```suggestion
    // and an issue will be generated. The unit is in megabytes.
```
?

**Review by FrankeLukas — APPROVED** [review]
Untested but changes make sense to me.

File: src/main/java/de/tum/in/ase/parser/strategy/StaticCodeAnalysisTool.java [hunk 2]
```diff
@@ -40,4 +46,19 @@ public ParserStrategy getStrategy() {
         }
         return Optional.empty();
     }
+
+    /**
+     * Finds a took by its filename (the xml file generated by the tool).
```
Comment (FuchsDominik) [inline src/main/java/de/tum/in/ase/parser/strategy/StaticCodeAnalysisTool.java]:
```suggestion
     * Finds a tool by its filename (the xml file generated by the tool).
```

**Review by FuchsDominik — APPROVED** [review]
As far as I can tell, the code changes look reasonable.

**Review by krusche — APPROVED** [review]

**Pull request merged**